src/backend/distributed/commands/table.c

```diff
@@ -224,9 +224,12 @@ PostprocessCreateTableStmtPartitionOf(CreateStmt *createStatement, const
 		char *parentRelationName = generate_qualified_relation_name(parentRelationId);
 		bool viaDeprecatedAPI = false;
 
-		CreateDistributedTable(relationId, parentDistributionColumn,
-							   parentDistributionMethod, ShardCount,
-							   parentRelationName, viaDeprecatedAPI);
+		if (!(IsCitusTable(relationId) && (createStatement->if_not_exists)))
```

Comment:
This check doesn't look right to me.

It means we will distribute the table if it is not a Citus table OR there is no `IF NOT EXISTS` statement. What if there is an `IF NOT EXISTS` statement and the table exists but it is just a regular Postgres table (that is not a partition). Won't this code just try to distribute it?

Reply:
Before there was no check in the code, I put one specific for the issue.
I guess before that check, we would enter that part of the code in this condition:
```
/*
  * If a partition is being created and if its parent is a distributed
  * table, we will distribute this table as well.
  */
```
So something that is not a partition shouldn't enter here.

If the partition already exists but is not distributed: actually this should never be the case for a parent distributed table. As I have seen from the tests, the codebase ensures that all partitions of a distributed table are distributed as well.

Reply:
* We can move lines [222, 225] to scope of this if statement, since we don't need/use them unless this if condition is satisfied.
* This if condition is not very readable. Adding comments would be nice.
* For Ozan's crashing case, `PreprocessCreateTableStmtPartitionOf` skips the creation since `if_not_exists` is true and a table with this name is already created. However, for this function, we don't know whether it's just created in `PreprocessCreateTableStmtPartitionOf` or was already existing. So I haven't tested it but maybe not calling `PreprocessCreateTableStmtPartitionOf` in such cases might be a solution

Reply:
By the way, couldn't find any `PreprocessCreateTableStmtPartitionOf`

Reply:
Sorry, didn't check that. But still, the case is the same. We execute `CREATE TABLE` queries (and others) **before** the Postprocess function.